src/OpenRiaServices.Client/Framework/ValidationResultCollection.cs

```diff
@@ -115,20 +115,21 @@ private void OnCollectionChanged(IEnumerable<string> propertiesAffected)
 
             // Determine our new state
             this._hasErrors = (this.Count > 0);
-            this._propertiesInError = GetPropertiesInError(this).Distinct();
+            this._propertiesInError = new HashSet<string>(GetPropertiesInError(this)); // HashSet is used to make properties distinct
 
             // Call the notification method if the 'HasErrors' bit has changed
             if (this._hasErrors != origHasErrors)
             {
                 this.OnHasErrorsChanged();
             }
 
-            // Find what properties were: in error but aren't any longer, newly in error, affected by the change
-            IEnumerable<string> noLongerInError = origPropertiesInError.Except(this._propertiesInError);
-            IEnumerable<string> newlyInError = this._propertiesInError.Except(origPropertiesInError);
-
-            // Get the combined list of properties affected.  The Union gives a Distinct result.
-            IEnumerable<string> allPropertiesAffected = noLongerInError.Union(newlyInError).Union(propertiesAffected);
+            // Get the combined list of properties affected.  
+            // SymmetricExceptWith - to get properties in error but aren't any longer or those newly in error
+            // add all affected by the change
+            // PERF: origPropertiesInError is already a HashSet which we could potentially use (Saves 1/8 of time and <20% of allocations)
```

Comment:
I did not dare to do this change since it would rincrease risk of subtle "bugs" in case an entity is concurrently changed on 2 different threads (but the end result of wrong names if both modify same hashset might not be "that bad").

The saving would only be about ~100ms for 1.5 million property setters so not a big deal

```suggestion
```

Reply:
I think that we can remove the comment, since if the bennefit is not that great and it is a bit risky I think it is not worth it